src/openloco/windows/IndustryWindow.cpp

```diff
@@ -0,0 +1,1064 @@
+                if (i < industry->history_size[1] - 1)
+                {
+                    if (yPos1 < self->y + self->height - 7)
+                    {
+                        if (yPos2 < self->y + self->height - 7)
+                        {
+                            gfx::draw_line(dpi, xPos, yPos1, xPos + 1, yPos2, colour::get_shade(self->colours[1], 7));
+                        }
+                    }
+                }
+                month--;
+                if (month < 0)
+                {
+                    month = 11;
+                    year--;
+
+                    yearSkip++;
+                    if (yearSkip >= 3)
+                        yearSkip = 0;
+                }
+            }
+        }
+
+        // 0x004565B5
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameIndustryPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_industry:
+                case common::widx::tab_production:
+                case common::widx::tab_production_2:
+                case common::widx::tab_transported:
+                    common::switchTab(self, widgetIndex);
+                    break;
+            }
+        }
+
+        // 0x004565FF
+        static void on_resize(window* self)
+        {
+            {
+                self->set_size(minWindowSize, maxWindowSize);
+            }
+        }
+
+        static void initEvents()
+        {
+            events.draw = draw;
+            events.on_mouse_up = on_mouse_up;
+            events.on_resize = on_resize;
+            events.on_update = common::update;
+            events.prepare_draw = prepare_draw;
+            events.text_input = common::text_input;
+        }
+    }
+
+    namespace transported
+    {
+        static const gfx::ui_size_t windowSize = { 300, 127 };
+
+        static widget_t widgets[] = {
+            commonWidgets(300, 126, string_ids::title_statistics),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x00456665
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            common::repositionTabs(self);
+        }
+
+        // 0x00456705
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            auto industry = industrymgr::get(self->number);
+            auto industryObj = objectmgr::get<industry_object>(industry->object_id);
+            auto totalReceivedCargo = industryObj->received_cargo[0] & industryObj->received_cargo[1] & industryObj->received_cargo[2];
```

Comment:
This is really just a cheap way of checking that at least one of the received_cargo s is not null. I think making the name `canReceiveCargo` and making it a bool would be better. Same principle goes for the produced cargo.